From 57211ecd55da396a703c0c8ed147f11bff6a89e3 Mon Sep 17 00:00:00 2001
From: Jose Carlos Venegas Munoz <jos.c.venegas.munoz@intel.com>
Date: Mon, 13 Apr 2015 11:26:36 -0500
Subject: [PATCH 103/114] fbcon: enable no blink by default

Author:    Arjan van de Ven <arjan@linux.intel.com>
---
 drivers/video/fbdev/core/fbcon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/core/fbcon.c b/drivers/video/fbdev/core/fbcon.c
index 929ca472c524..fa5947f4d440 100644
--- a/drivers/video/fbdev/core/fbcon.c
+++ b/drivers/video/fbdev/core/fbcon.c
@@ -148,7 +148,7 @@ static const struct consw fb_con;
 
 static int fbcon_set_origin(struct vc_data *);
 
-static int fbcon_cursor_noblink;
+static int fbcon_cursor_noblink = 1;
 
 #define divides(a, b)	((!(a) || (b)%(a)) ? 0 : 1)
 
-- 
2.16.1

